app/src/main/java/com/andrew/tasky/agenda/data/util/UriByteConverter.kt

```diff
@@ -0,0 +1,27 @@
+package com.andrew.tasky.agenda.data.util
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.net.Uri
+import java.io.ByteArrayOutputStream
+
+class UriByteConverter(
+    val appContext: Context
+) {
+    fun uriToByteArray(uri: Uri, targetSize: Int): ByteArray {
+        val imageByte = appContext.contentResolver.openInputStream(uri).use {
+            it!!.readBytes()
+        }
+        val bitmap = BitmapFactory.decodeByteArray(imageByte, 0, imageByte.size)
+        var quality = 100
+        var result = imageByte
+        while (result.size > targetSize && quality > 0) {
+            val stream = ByteArrayOutputStream()
+            bitmap.compress(Bitmap.CompressFormat.JPEG, quality, stream)
+            result = stream.toByteArray()
+            quality -= 5
+        }
+        return result
+    }
```

Comment:
The code for compressing the image should also rather go in a separate class, for example `BitmapCompressor` or so. In your case that's quite a side effect of the `uriToByteArray` function. Imagine that function would come from a library and would just silently compress the read image without you being able to change that behavior. That's one reason why we stick to the single-responsibility principle, so that we're really flexible with how we combine different classes.